plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/PluginVersionChecker.kt

```diff
@@ -0,0 +1,70 @@
+// Copyright 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains
+
+import com.intellij.ide.ApplicationInitializedListener
+import com.intellij.ide.plugins.IdeaPluginDescriptor
+import com.intellij.ide.plugins.PluginEnabler
+import com.intellij.notification.NotificationAction
+import com.intellij.notification.NotificationType
+import com.intellij.notification.SingletonNotificationManager
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.application.ex.ApplicationManagerEx
+import kotlinx.coroutines.CoroutineScope
+import software.aws.toolkits.core.utils.tryOrNull
+import software.aws.toolkits.jetbrains.core.plugin.PluginUpdateManager
+
+class PluginVersionChecker : ApplicationInitializedListener {
+    override suspend fun execute(asyncScope: CoroutineScope) {
+        val core = AwsToolkit.PLUGINS_INFO.get(AwsPlugin.TOOLKIT) ?: return
+        val mismatch = AwsToolkit.PLUGINS_INFO.values.filter { it.version != core.version }
+
+        if (mismatch.isEmpty()) {
+            return
+        }
+
+        val updated = mismatch.filter {
+            val descriptor = it.descriptor as? IdeaPluginDescriptor ?: return@filter false
+
+            tryOrNull {
+                PluginUpdateManager.getUpdate(descriptor)?.install()
+                true
+            } ?: false
+        }
+
+        if (updated.isNotEmpty() && !ApplicationManager.getApplication().isUnitTestMode) {
+            ApplicationManagerEx.getApplicationEx().restart(true)
+            return
```

Comment:
is this return required since we are restarting?

Reply:
good practice to signify no further action